src/kernel-install/40-early-cpio.install

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+# -*- mode: shell-script; indent-tabs-mode: nil; sh-basic-offset: 4; -*-
+# ex: ts=8 sw=4 sts=4 et filetype=sh
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# This file is part of systemd.
+#
+# systemd is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# systemd is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with systemd; If not, see <https://www.gnu.org/licenses/>.
+
+set -e
+
+COMMAND="${1:?}"
+ENTRY_DIR_ABS="${3:?}"
+
+case "$COMMAND" in
+    remove)
+        [ "$KERNEL_INSTALL_LAYOUT" = "bls" ] || exit 0
+
+        for ucode in "${ENTRY_DIR_ABS}"/early-cpio*; do
+            [ -f "$ucode" ] || continue
+            [ "$KERNEL_INSTALL_VERBOSE" -gt 0 ] && echo "Removing $ucode"
+            rm -f "$ucode"
+        done
+        ;;
```

Comment:
Is a `remove` action really necessary? For `bls` this exit early anyway, since it the `90-loaderentry.install` should clean everything and for `uki` the microcode should be embedded in the UKI, which will be cleaned up by `90-uki-copy.install`.

Reply:
> As far as I understand, 90-loaderentry.install doesn't remove anything apart from the .conf file

Sorry, my bad, I had this confused. I vaguely remembered a PR that was doing some hooks to count whether things were still needed.

But still, I don't think this plugin should touch things after the fact and directories for additional files should only be cleaned up together.